pkg/apis/kubermatic/v1/settings.go

```diff
@@ -129,9 +129,27 @@ type SettingSpec struct {
 	// StaticLabels are a list of labels that can be used for the clusters.
 	StaticLabels []StaticLabel `json:"staticLabels,omitempty"`
 
+	// Annotations are the settings for the annotations in KKP UI.
+	Annotations AnnotationSettings `json:"annotations,omitempty"`
+
 	// TODO: Datacenters, presets, user management, Google Analytics and default addons.
 }
 
+// AnnotationSettings is the settings for the annotations.
+type AnnotationSettings struct {
+	// +kubebuilder:default:={"kubectl.kubernetes.io/last-applied-configuration", "kubermatic.io/initial-application-installations-request", "kubermatic.io/initial-machinedeployment-request"}
+
+	// HiddenAnnotations are the annotations that are hidden from the user in the UI.
+	// +optional
+	HiddenAnnotations []string `json:"hiddenAnnotations,omitempty"`
```

Comment:
Note for the reviewer:
Maybe we can just call it `hidden`, but to me it sounded a bit ambiguous